src/core/stream_set.c

```diff
@@ -191,8 +193,10 @@ QuicStreamSetReleaseStream(
     uint8_t Flags = (uint8_t)(Stream->ID & STREAM_ID_MASK);
     QUIC_STREAM_TYPE_INFO* Info = &StreamSet->Types[Flags];
 
+    CxPlatDispatchLockAcquire(&StreamSet->TypesLock);
```

Comment:
I _think_ this is the only other synchronization we'd need (besides in `QuicStreamSetNewLocalStreamID` below), but it's a bit tricky.